Desktop/modules/description/entrybase.h

```diff
@@ -23,72 +23,82 @@ class EntryBase : public DescriptionChildBase
-	Q_PROPERTY(bool			requiresData	READ requiresData	WRITE setRequiresData	NOTIFY requiresDataChanged	)
-	Q_PROPERTY(bool			enabled			READ enabled		WRITE setEnabled		NOTIFY enabledChanged		) //Hmm, this already exists in QQuickItem, maybe a problem?
-	Q_PROPERTY(bool			debug			READ debug			WRITE setDebug			NOTIFY debugChanged			)
+	Q_PROPERTY(QString		menu			READ menu				WRITE setMenu				NOTIFY menuChanged				)
+	Q_PROPERTY(QString		title			READ title				WRITE setTitle				NOTIFY titleChanged				)
+	Q_PROPERTY(QString		func			READ function			WRITE setFunction			NOTIFY functionChanged			)
+	Q_PROPERTY(QString		funcInternal	READ functionInternal	WRITE setFunctionInternal	NOTIFY functionInternalChanged	)
+	Q_PROPERTY(QString		icon			READ icon				WRITE setIcon				NOTIFY iconChanged				)
+	Q_PROPERTY(QString		qml				READ qml				WRITE setQml				NOTIFY qmlChanged				)
+	Q_PROPERTY(EntryType	entryType		READ entryType										NOTIFY entryTypeChanged			) //Entry type can only be set in constructor, to keep things manageable
+	Q_PROPERTY(bool			requiresData	READ requiresData		WRITE setRequiresData		NOTIFY requiresDataChanged		)
+	Q_PROPERTY(bool			enabled			READ enabled			WRITE setEnabled			NOTIFY enabledChanged			) //Hmm, this already exists in QQuickItem, maybe a problem?
+	Q_PROPERTY(bool			debug			READ debug				WRITE setDebug				NOTIFY debugChanged				)
+	Q_PROPERTY(bool			hasWrapper		READ hasWrapper			WRITE setHasWrapper			NOTIFY hasWrapperChanged		)
 
 public:
 	enum class EntryType {unknown, separator, groupTitle, analysis};
 	Q_ENUM(EntryType);
 
 	EntryBase(EntryType entryType);
 
-	QString		menu()			const { return _menu;			}
-	QString		title()			const { return _title;			}
-	QString		function()		const { return _function;		}
-	QString		icon()			const { return _icon;			}
-	EntryType	entryType()		const { return _entryType;		}
-	bool		requiresData()	const { return _requiresData;	}
-	bool		enabled()		const { return _enabled;		}
-	QString		qml()			const { return _qml;			}
-	bool		debug()			const { return _debug;			}
-	QString		toString()		const;
-	bool		shouldBeAdded()	const;
+	QString		menu()				const { return _menu;				}
+	QString		title()				const { return _title;				}
+	QString		function()			const { return _function;			}
+	QString		functionInternal()	const { return _functionInternal;	}
+	QString		icon()				const { return _icon;				}
+	EntryType	entryType()			const { return _entryType;			}
+	bool		requiresData()		const { return _requiresData;		}
+	bool		enabled()			const { return _enabled;			}
+	QString		qml()				const { return _qml;				}
+	bool		debug()				const { return _debug;				}
+	bool		hasWrapper()		const { return _hasWrapper;			}
+	QString		toString()			const;
+	bool		shouldBeAdded()		const;
 
 	///This function is a stopgap and these two classes must be merged together later
-	AnalysisEntry * convertToAnalysisEntry(bool requiresDataDefault) const;
+	AnalysisEntry * convertToAnalysisEntry(bool requiresDataDefault, bool hasWrappers) const;
```

Comment:
Why does it have this flag?

entryBase already knows whether it has a wrapper...